src/pygeosx/unitTests/dataRepositoryTests/test_pygeosxDataRepository.py

```diff
@@ -0,0 +1,181 @@
+        """
+        group = geosx_problem.get_group('Events', None)
+        assert isinstance(group, pygeosx.Group)
+        t = group.get_wrapper('time').value()[0]
+        assert isinstance(t, float)
+
+    # yapf: disable
+    @pytest.mark.parametrize('target_key, target_shape, target_type',
+                             [('Events/time', (1, ), float),
+                              ('Events/cycle', (1, ), np.int32),
+                              ('Events/python/target', 21, str),
+                              (node_manager + 'ReferencePosition', (8, 3), float),
+                              (node_manager + 'ghostRank', (8, ), np.int32),
+                              (cell_blocks + 'elementVolume', (1, ), float)])
+    def test_wrapper_read(self, target_key, target_shape, target_type, geosx_problem):
+        """
+        Test whether wrappers can be read, and whether their shape and value types are correct
+
+        Args:
+            target_key (str): Key value for the target wrapper
+            target_shape (tuple): Expected shape for the target wrapper
+            target_type (dtype): Expected type for the wrapper elements
+            geosx_proplem (pygeosx.problem): The problem handle
+        """
+        # yapf: enable
+        s = 0
+        val = 0
+        x = geosx_problem.get_wrapper(target_key).value()
+        if hasattr(x, "to_numpy"):
+            x = x.to_numpy()
+
+        if isinstance(x, np.ndarray):
+            s = np.shape(x)
+            Ia = tuple([0 for y in s])
+            val = x[Ia]
+        else:
+            s = len(x)
+            val = x
+
+        assert s == target_shape
+        assert isinstance(val, target_type)
+
+    # yapf: disable
+    @pytest.mark.parametrize('target_key',
+                             [('Events/time'),
+                              ('Events/cycle'),
+                              (node_manager + 'ReferencePosition'),
+                              (node_manager + 'ghostRank'),
+                              (cell_blocks + 'elementVolume')])
+    def test_wrapper_write(self, target_key, geosx_problem):
+        """
+        Test writing to the wrappers
+
+        Args:
+            target_key (str): Key value for the target wrapper
+            geosx_proplem (pygeosx.problem): The problem handle
+        """
+        # yapf: enable
+        # Access and record the initial value
+        x = geosx_problem.get_wrapper(target_key).value()
+        if hasattr(x, "set_access_level"):
+            x.set_access_level(pylvarray.MODIFIABLE, pylvarray.CPU)
+        if hasattr(x, "to_numpy"):
+            x = x.to_numpy()
+        x_init = x.copy()
+
+        # Modify the value
+        x += 1
+        del x
+
+        # Get another copy of the wrapper
+        x = geosx_problem.get_wrapper(target_key).value()
+        if hasattr(x, "to_numpy"):
+            x = x.to_numpy()
+
+        # Check to see if the value changed appropriately
+        dx = x_init + 1 - x
```

Comment:
I think it would be safer to use the absolute value of the difference, in case the discrepancy goes in the other direction

Reply:
Missed this one.   Yes, at a minimum it should be abs(sum(values))